airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/PostgresSourceOperations.java

```diff
@@ -398,9 +407,16 @@ public PostgresType getDatabaseFieldType(final JsonNode field) {
         // It should not be converted to base64 binary string. So it is represented as JDBC VARCHAR.
         // https://www.postgresql.org/docs/14/datatype-binary.html
         case "bytea" -> PostgresType.VARCHAR;
+        case "numeric" -> {
+          if (field.get(INTERNAL_DECIMAL_DIGITS) != null && field.get(INTERNAL_DECIMAL_DIGITS).asInt() == 0) {
+            yield BIGINT;
+          } else {
+            yield defaultType;
+          }
+        }
         case TIMESTAMPTZ -> PostgresType.TIMESTAMP_WITH_TIMEZONE;
         case TIMETZ -> PostgresType.TIME_WITH_TIMEZONE;
-        default -> PostgresType.valueOf(field.get(INTERNAL_COLUMN_TYPE).asInt(), POSTGRES_TYPE_DICT);
+        default -> defaultType;
```

Comment:
one unintended side effect of moving `PostgresType.valueOf(field.get(INTERNAL_COLUMN_TYPE).asInt(), POSTGRES_TYPE_DICT);` further up this method is that it will be executed for all cases and not just the default. Do we know if all `INTERNAL_COLUMN_TYPE` are mapped to `POSTGRES_TYPE_DICT`? As in will is there a chance we run into a `KeyNotFoundException`?

Reply:
remembered KeyNotFoundException is .NET C# land :) but the question still remains for 1:1 mapping of INTERNAL_COLUMN_TYPE to POSTGRES_TYPE_DICT and what happens if `defaultType` comes to be null?

Reply:
Ah, good call. Will move it back to the original level and investigate if we could potentially running into a missing key problem.